src/CMakeLists.txt

```diff
@@ -76,6 +76,10 @@ function(valhalla_module)
   set(library valhalla-${MODULE_NAME})
   add_library(${library} OBJECT ${MODULE_SOURCES} ${MODULE_HEADERS})
   add_library(valhalla::${MODULE_NAME} ALIAS ${library})
+
+  # Generate source groups so the files are properly sorted in IDEs like Xcode.
+  source_group(TREE "${PROJECT_SOURCE_DIR}/" FILES ${MODULE_SOURCES} ${MODULE_HEADERS})
```

Comment:
if this problem is limited to xcode perhaps we should wrap this in: 

    if(CMAKE_GENERATOR STREQUAL Xcode)

or does this have a benefit in other IDEs?

Reply:
It should also work in Visual Studio

Reply:
it looks like i wrongly assumed the build failure was caused by this. it looks kind of like a github outage borked the windows builds.. im trying a re-run

Reply:
actually i take that back, looks like it is failing the build: https://dev.azure.com/valhalla1/valhalla/_build/results?buildId=2497&view=logs&j=1258c03f-82ba-5c36-5c40-0722b4c18a9a&t=b7ad8864-1b6e-57a6-c257-820c8f60e233

```
CMake Error at src/CMakeLists.txt:81 (source_group):
  source_group ROOT: D:/a/1/s is not a prefix of file:
  D:/a/build/src/valhalla/proto/api.pb.cc
Call Stack (most recent call first):
  proto/CMakeLists.txt:13 (valhalla_module)


CMake Error at src/CMakeLists.txt:81 (source_group):
  source_group ROOT: D:/a/1/s is not a prefix of file:
  D:/a/build/src/baldr/date_time_africa.h
Call Stack (most recent call first):
  src/baldr/CMakeLists.txt:94 (valhalla_module)


CMake Error at src/CMakeLists.txt:81 (source_group):
  source_group ROOT: D:/a/1/s is not a prefix of file:
  D:/a/build/src/odin/locales.h
Call Stack (most recent call first):
  src/odin/CMakeLists.txt:28 (valhalla_module)
```

Its complaining about source files that we generate as part of the build.. I dont know enough about the cmake feature to have a clue as to why that would be.

Reply:
Azure Pipelines doesn't use a build directory that is a _subdirectory_ of the source directory. However, the `source_group` code made that assumption. I've changed the PR so that generated files, which are in the _build_ directory, are in a `Generated Files` folder, while the source code files are in the respective folders.